docs/blockstates/states.md

```diff
@@ -0,0 +1,79 @@
+
+`IBlockState` is the new "block + meta" of 1.8. Within it, it carries information about a Block, its properties, and a *value* for each of those properties. For example, an `IBlockState` of a wooden log contains the Block "Blocks.log" or "Blocks.log2", the *properties* AXIS and VARIANT, and *values* for those properties, e.g. `Axis.Y` and `BlockPlanks.WoodType.BIRCH`.
+You can acquire one with `world.getBlockState(BlockPos)`
+To test the value of a property, use IBlockState.getValue(<property>). So for our log, if we wanted to see if it was birch we would do `world.getBlockState(pos).getValue(VARIANT) == BlockPlanks.WoodType.BIRCH`.
+To get the IBlockState object for another combination of properties and valuee, use `withProperty(IProperty<T>, T)`. This will return a different IBlockState object with the values you requested.
+
+NOTE: All IBlockState objects are generated by Minecraft on startup. For each unique set of block+properties+values, there is exactly one IBlockState. Thus it is perfectly safe and recommended to compare IBlockStates using reference equality (==) ! 
+
+In the constructor of your block, it is highly recommended you set the default blockstate. You simply do this by calling `setDefaultState(blockState.getBaseState().withProperty(...))`. Be sure to do this for ALL of your `IProperty`'s, or the values of the ones you skip will be undefined.
+
+Saving to meta and coming back from meta
+------------------------
+
+If your Block uses no `IProperty`'s, you don't need to override these two methods. Otherwise, you MUST or else Minecraft will crash on startup.
+Override getMetaFromState and getStateFromMeta. These two methods are where all the ugly bitmasking/bitpacking/bit testing will take place. Note that since the save format has not changed, you are still limited to 16 meta values (0 to 15 inclusive).
+
+WARNING: Your getMetaFromState and getStateFromMeta methods MUST be one to one! In other words, the same set of properties and values must map to the same meta value and back. Failing to do this, unfortunately, WON'T cause a crash. It'll just cause everything to behave extremely weirdly.
+
+IBlockState is more than just Block and meta
+------------------------
+
+This isn't the end of the story, though. 
+If you've ever hovered over a vanilla fence with F3 activated, you might have noticed the boolean properties north, south, west, and east with. Sharper minds might notice that fences don't save their connection to meta, so what's going on here?
+It turns out, `IBlockState`s are able to hold values for properties that aren't saved to meta! You register them in createBlockState as usual.
+Override `getActualState` in your block class. Here, fences and redstones check their sides for connections, and modify the state appropriately. 
+
+Note: The result of `getActualState` is only considered for rendering purposes. Querying `world.getBlockState()` will give you the `IBlockState` representing the saved metadata.
```

Comment:
Admonition. Also, this is not true. Actual state is passed into most block hook methods, for example. It is *not* purely for rendering. That's extended states.